### PR TITLE
feat(#45): discount calculation form

### PR DESCRIPTION
**Pull request by mjkeaton**
Adds a `DiscountForm` component with two modes: "Gross to net" or "Net to gross".

Relates to #45.

## 💡 How to Test

Please provide clear instructions on how reviewers can test your changes:

1. `npm run storybook`
2. visit http://localhost:6006/?path=/docs/element-discountform--docs
3. verify discount form ("Gross to net" & "Net to gross")

Please also verify the tests, which should follow [the formula described in this document](https://docs.google.com/spreadsheets/d/1YGI32u4QEVDLD8SksdvDjaQ4z3WjL1GrIHgVUxNaQqI/edit?gid=0#gid=0):
- `discount-util.test.ts`
- `DiscountForm.test.tsx`

---

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -11,6 +11,9 @@ export default {
         sans: ['Geist', 'system-ui', 'sans-serif'],
         mono: ['Geist Mono', 'monospace'],
       },
+      fontSize: {
+        xxs: '0.625rem',
```
Comment (cryptosalomao) [inline tailwind.config.js:15]: The `xxs` size is already used by fonts with a size of 0.75rem/12px. When needed, we must hard code font-[10px].

File: src/utils/index.ts [hunk 2]
```diff
@@ -34,3 +34,26 @@ export const truncateFileName = (name: string) => {
...
+const formatDMY = (date: Date, locale: string): string => {
+  const year = new Intl.DateTimeFormat(locale, { year: "2-digit" }).format(date);
+  const month = new Intl.DateTimeFormat(locale, { month: "short" }).format(date);
+  const day = new Intl.DateTimeFormat(locale, { day: "2-digit" }).format(date);
+  return `${day}-${month}-${year}`;
+};
+
+export const formatDate = (date: Date, locale: string): string => {
+  return formatDMY(date, locale);
+};
```
Comment (cryptosalomao) [inline src/utils/index.ts:47]: Two things:

- maybe move the date specific utils to their own util file? something like `utils/date.ts`
- what is the case for wrapping the caller of `formatDMY` on another function?

File: src/utils/index.ts [hunk 3]
```diff
@@ -34,3 +34,26 @@ export const truncateFileName = (name: string) => {
...
+  if (str === undefined) return undefined;
+  const parsed = parseFloat(str);
+  return isNaN(parsed) || !isFinite(parsed) ? undefined : parsed;
+};
+
+export const parseIntSafe = (str: string | undefined) => {
+  if (str === undefined) return undefined;
+  const parsed = parseInt(str, 10);
+  return isNaN(parsed) || !isFinite(parsed) ? undefined : parsed;
+};
```
Comment (cryptosalomao) [inline src/utils/index.ts:59]: move these to their own utils file? `utils/number.ts`

**Review by cryptosalomao — COMMENTED** [review]
All good so far, with a few points to be addressed. One important thing is to move all the component related files to its own directory under the `components/` folder.

File: src/utils/index.ts (hunk 2 above)
Comment (mjkeaton) [reply src/utils/index.ts:47]: >     * maybe move the date specific utils to their own util file? something like `utils/date.ts`

Yes!
 
>     * what is the case for wrapping the caller of `formatDMY` on another function?

Idea was to indicate the possibility of adding format functions or switching the default format easily if needed. So.. you are right–it is not needed. I will simplify.

**Review by cryptosalomao — APPROVED** [review]

**Pull request merged**